addon/components/document-list-item.js

```diff
@@ -20,6 +38,67 @@ export default class DocumentListItemComponent extends Component {
     return classes.join(" ");
   }
 
+  get iconStyle() {
+    let iconClass = "file-alt";
+    let color = "#6c757d";
+
+    if (this.mimeType) {
+      if (this.mimeType.startsWith("image/")) {
+        iconClass = "file-image";
+        color = "#28a745";
+      } else if (this.mimeType === "application/pdf") {
+        iconClass = "file-pdf";
+        color = "#dc3545";
+      } else if (
+        this.mimeType.includes("word") ||
+        [
+          "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+          "application/msword",
+        ].includes(this.mimeType)
+      ) {
+        iconClass = "file-word";
+        color = "#2b579a";
+      } else if (
+        this.mimeType.includes("excel") ||
+        this.mimeType.includes("spreadsheet") ||
+        [
+          "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+          "application/vnd.ms-excel",
+        ].includes(this.mimeType)
+      ) {
+        iconClass = "file-excel";
+        color = "#217346";
+      } else if (
+        this.mimeType.includes("powerpoint") ||
+        this.mimeType.includes("presentation") ||
+        this.mimeType ===
+          "application/vnd.openxmlformats-officedocument.presentationml.presentation" ||
+        this.mimeType === "application/vnd.ms-powerpoint"
+      ) {
+        iconClass = "file-powerpoint";
+        color = "#d24726";
+      } else if (
+        this.mimeType === "application/vnd.ms-outlook" ||
+        this.mimeType === "message/rfc822"
+      ) {
+        iconClass = "envelope";
+        color = "#0078d4";
+      } else if (
+        ["image/vnd.dwg", "application/acad", "application/x-dwg"].includes(
+          this.mimeType,
+        )
+      ) {
+        iconClass = "file";
+        color = "#ffc107";
+      }
+    }
+
+    return {
+      iconClass,
+      color,
+    };
+  }
```

Comment:
Please consider that the logic for getting the right icon already exists and should be reused:

https://github.com/projectcaluma/ember-alexandria/blob/fd37bff0966729c439b8825f7200966443360a17/addon/models/file.js#L50-L67

Reply:
right, this method was just a draft, wanted to rework it anyway but that helps, thanks!